src/components/views/settings/KeyBackupPanel.js

```diff
@@ -0,0 +1,240 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+
+import sdk from '../../../index';
+import MatrixClientPeg from '../../../MatrixClientPeg';
+import { _t } from '../../../languageHandler';
+import Modal from '../../../Modal';
+
+export default class KeyBackupPanel extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this._startNewBackup = this._startNewBackup.bind(this);
+        this._deleteBackup = this._deleteBackup.bind(this);
+        this._verifyDevice = this._verifyDevice.bind(this);
+        this._onKeyBackupStatus = this._onKeyBackupStatus.bind(this);
+        this._restoreBackup = this._restoreBackup.bind(this);
+
+        this._unmounted = false;
+        this.state = {
+            loading: true,
+            error: null,
+            backupInfo: null,
+        };
+    }
+
+    componentWillMount() {
+        this._loadBackupStatus();
+
+        MatrixClientPeg.get().on('keyBackupStatus', this._onKeyBackupStatus);
+    }
+
+    componentWillUnmount() {
+        this._unmounted = true;
+
+        if (MatrixClientPeg.get()) {
+            MatrixClientPeg.get().removeListener('keyBackupStatus', this._onKeyBackupStatus);
+        }
+    }
+
+    _onKeyBackupStatus() {
+        this._loadBackupStatus();
+    }
+
+    async _loadBackupStatus() {
+        this.setState({loading: true});
+        try {
+            const backupInfo = await MatrixClientPeg.get().getKeyBackupVersion();
+            const backupSigStatus = await MatrixClientPeg.get().isKeyBackupTrusted(backupInfo);
+            if (this._unmounted) return;
+            this.setState({
+                backupInfo,
+                backupSigStatus,
+                loading: false,
+            });
+        } catch (e) {
+            console.log("Unable to fetch key backup status", e);
+            if (this._unmounted) return;
+            this.setState({
+                error: e,
+                loading: false,
+            });
+            return;
+        }
+    }
+
+    _startNewBackup() {
+        const CreateKeyBackupDialog = sdk.getComponent("dialogs.keybackup.CreateKeyBackupDialog");
+        Modal.createTrackedDialog('Key Backup', 'Key Backup', CreateKeyBackupDialog, {
+            onFinished: () => {
+                this._loadBackupStatus();
+            },
+        });
+    }
+
+    _deleteBackup() {
+        const QuestionDialog = sdk.getComponent("dialogs.QuestionDialog");
+        Modal.createTrackedDialog('Delete Backup', '', QuestionDialog, {
```

Comment:
minor nit: doublequotes for strings, iirc? rather than flipping back and forth in the same method

Reply:
Our code style says single by default so I've changed them to be consistent that way (apart from the longer, more English-y ones that are liable to contain apostrophes).